### DISCUSSION TITLE
Is there a more granular way to track changes?

### DISCUSSION THREAD

**Discussion by SarthakSingh31** [bevyengine/bevy#2526] [Q&A] [answered]
I have a struct:
```rust
pub enum GameScene {
    Menu,
    Build {
        loaded: bool,
        builds: Vec<unit::build::Build>,
        parts: Option<unit::build::AllParts>,
        textures: Vec<Handle<Texture>>,
    },
    Fight { level: level::Level, mode: FightMode },
    GameOver,
}
```
I can track changes on `GameScene` but can I track which one of the fields `GameScene::Build::{loaded/builds/parts/textures}` changed?

I don't want to manually track the change because then I will have to use `ResMut<GameScene>` everywhere to update the internal tracking. `GameScene::is_changed` will also become meaningless.

**Comment by mockersf** [comment, marked as answer]
Bevy change detection is on resource level, so no, no built in more granular way...

That said:
- you could split your resource into smaller ones, grouping them by things that make sense to be used together for change detection
- you would probably need a `ResMut<GameScene>` to update change tracking only when you update the resource, so you should already have a `ResMut`?
- you could look into rust patterns for interior mutability for your change detection if that's an issue

**Reply by SarthakSingh31** [reply]
Interior mutability looks promising. I will try it.
